developers/contributor-guide/current/weaviate-contributors-guide/suggesting-enhancements.md

```diff
@@ -0,0 +1,106 @@
+
+How suggesting enhancements will help you?
+
+* Receive timely feedback from engineers on requests.
+
+* Learn more about upcoming releases.
+
+* Increase your involvement over feature development
+
+This section walks you through the process of submitting an enhancement for an addition to Weaviate core, such as a brand-new feature or a small change to an already-existing feature. 
+
+Following these guidelines can help you make a better suggestion, and make it easier for the maintainers and the community to understand your proposal and find other similar suggestions.
+
+You can still file an issue if you have an idea for a new feature but aren't sure where it should go. When filling out the issue template, be sure to include the steps you would take if the requested feature were available.
+
+We recommend you to check the Weaviate Module System to get better understanding of how Weaviate works.
+
+* [Overview and High-Level Architecture](../weaviate-module-system/overview.html)
+* [Code-Level Architecture](../weaviate-module-system/architecture.html)
+
+Be sure to checkout the [folder structure](#folder-structure) to get high level overview of the repository.
+
+## How do I make a (good) suggestion for improvement?
+
+Enhancement suggestions are tracked as GitHub issues. Check if there's already a WIP (work in progress) issue which describes that enhancement. If not, create an issue in the repository using the template and provide the following information:
+
+* **Use a clear and descriptive title** for the issue to identify the suggestion.
+* Give as many specifics as you can in a **step-by-step description** of the suggested improvement.
+* **Add specific examples** to demonstrate the steps. If possible, include code snippets in Markdown format.
+* **Describe the current behavior** and then **explain which behaviour you were expecting and why**.
+* **Include images, animated GIFs, or video links** that can be used to illustrate the steps or highlight the area of Weaviate that the suggestion relates to.
+* **Explain why this enhancement would be useful** to most Weaviate users.
+* Specify which **version of Weaviate** you're using. Check the version in your `docker-compose.yml` file.
+
+In order to get your requests into these prioritization meetings, make sure they are clearly written with details about what you’re trying to accomplish (Your use cases) , your current workaround (if you have one), what problems you’re having, and any other details that will help build a case for why this is an important request to you. 
+
+## Folder Structure
+
+To get a better understanding of the Weaviate Core project is organised, check out the [autogenerated folder structure](https://pkg.go.dev/github.com/semi-technologies/weaviate){:target="_blank"}, which should give you an instight into the purpose of each folder.
```

Comment:
Should be omitted IMO

Reply:
This is something I had a chat with Etienne a while ago.
Tbh. there isn't much harm in us trying to explain the folder structure of a project.
In some cases, this can be quite helpful.

Or do you think it is weird to have that in here?

Reply:
I'm not strongly opposed to this, we can leave it 🙂 one thing I would suggested is changing `autogenerated folder structure` to `Go reference docs`

Reply:
perfect! this is great